README.md

```diff
@@ -45,16 +45,16 @@ The following is an extended example with all possible options available for thi
     # See the `pathspec`-documentation for git
     # - https://git-scm.com/docs/git-add#Documentation/git-add.txt-ltpathspecgt82308203
     # - https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefpathspecapathspec
-    file_pattern: src/*.js tests/*.js *.php
+    file_pattern: 'src/*.js tests/*.js *.php'
```

Comment:
The Action would parse this as a string and would get `src/*.js tests/*.js *.php` as an input. And not `src/*.js`, `tests/*.js` and `*.php`. This unfortunately wouldn't work.

Reply:
Hey Stefan. Hmmm, interestingly, I was running into string token issues on my build. I'll play around a little bit and close the PR if I find out I was wrong. Certainly there's a reason why most of the examples don't use string quotes... :)

probably jumped the gun on the PR.

Reply:
So here's what I'm getting:

![image](https://user-images.githubusercontent.com/50302144/147689510-84d9122e-8497-4de3-8048-df51fad936f6.png)

Am I doing something stupid? :p

Reply:
Mh. I guess `file_pattern` is on line 71.

Could you give it a go by adding the following option?

```
disable_globbing: true
```

This was added in response to https://github.com/stefanzweifel/git-auto-commit-action/issues/153.
Or did your inital approach of putting the file pattern in quotes work?

Have to do some exploring with local tests, what the issue here really is.

Reply:
I understand the problem better now:  Adding `*.png *.md` to a YAML file will throw a parsing error. We can't wrap them in quotes, as the Action will then interpret them as strings.

I did the following in a demo repository, which seems to work as expected.

```yaml
- uses: stefanzweifel/git-auto-commit-action@v4
  with:
    disable_globbing: true
    file_pattern: ./*.png ./*.md
```

This will look for changed PNG and MD files in the root of your repository.

Link to my test repository: https://github.com/stefanzweifel/git-auto-commit-action-demo-app/blob/42c0c118a7254e5363d12039d4ac4eb2ae4cc5cc/.github/workflows/format_php.yml#L19-L23

Did that work for you?
If yes, I will update the README with an updated example – or feel free to update your pull request :)

Reply:
That's still just equivalent to putting `"./*.png ./*.md"` in quotes. You can use one of the online yaml to json converters to see this. However you do this, the pattern would be converted to a string. You need the bash script to expand that string into separate args.

Reply:
@ForbesLindesay You're right. `./*.png ./*.md` equals `"./*.png ./*.md"`. I think my comment just wanted to illustrate that you could write the same pattern without using quotes.

> You need the bash script to expand that string into separate args.

Did you have something like this in mind? https://github.com/stefanzweifel/git-auto-commit-action/pull/205/files
By expanding `file_pattern` into an array, we pass not `*.png` to the git command, but rather `foo.png bar.png`.

Any feedback or input is greatly appreciated.

Reply:
In #205 the long standing issues with file_pattern and quotes and finally been solved.
There are still edge cases, that have been documented in the README: https://github.com/stefanzweifel/git-auto-commit-action#custom-file_pattern-changed-files-but-seeing-working-tree-clean-nothing-to-commit-in-the-logs

I'm gonna merge this PR now, as the updated example better illustrates what's possible with the Action.